src/symbolize/mod.rs

```diff
@@ -69,26 +79,52 @@ impl Symbol {
 
     /// Returns the starting address of this function.
     pub fn addr(&self) -> Option<*mut c_void> {
-        self.inner.addr()
+        self.inner.addr().map(|p| p as *mut _)
     }
 
+    /// Returns the raw filename as a slice. This is mainly useful for `no_std`
+    /// environments.
+    pub fn filename_raw(&self) -> Option<BytesOrWideString> {
+        self.inner.filename_raw()
+    }
+
+
+    /// Returns the line number for where this symbol is currently executing.
+    ///
+    /// This return value is typically `Some` if `filename` returns `Some`, and
+    /// is consequently subject to similar caveats.
+    pub fn lineno(&self) -> Option<u32> {
+        self.inner.lineno()
+    }
+}
+
+#[cfg(feature = "std")]
+impl Symbol {
     /// Returns the file name where this function was defined.
     ///
     /// This is currently only available when libbacktrace is being used (e.g.
     /// unix platforms other than OSX) and when a binary is compiled with
     /// debuginfo. If neither of these conditions is met then this will likely
     /// return `None`.
+    #[cfg(not(windows))]
     pub fn filename(&self) -> Option<&Path> {
-        self.inner.filename()
+        use std::ffi::OsStr;
+        use std::os::unix::ffi::OsStrExt;
+
+        match self.filename_raw() {
+            Some(BytesOrWideString::Bytes(slice)) => {
+                Some(Path::new(OsStr::from_bytes(slice)))
+            }
+            _ => None,
+        }
     }
 
-    /// Returns the line number for where this symbol is currently executing.
-    ///
-    /// This return value is typically `Some` if `filename` returns `Some`, and
-    /// is consequently subject to similar caveats.
-    pub fn lineno(&self) -> Option<u32> {
-        self.inner.lineno()
+    /// Returns the file name where this function was defined.
+    #[cfg(windows)]
+    pub fn filename(&self) -> Option<&Path> {
+        self.inner.filename().map(Path::new)
```

Comment:
This is definitely a bummer, but unfortunately I don't know of a better way to do it! Next time we'll for sure make this return a `PathBuf`